openai-core/src/commonMain/kotlin/com.aallam.openai.api/run/AssistantStreamEvent.kt

```diff
@@ -0,0 +1,251 @@
+    public val serializer: KSerializer<*>
+) {
+
+    /**
+     * Occurs when a new thread is created.
+     */
+    THREAD_CREATED("thread.created", Thread::class, Thread.serializer()),
+
+    /**
+     * Occurs when a new run is created.
+     */
+    THREAD_RUN_CREATED("thread.run.created", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run moves to a queued status.
+     */
+    THREAD_RUN_QUEUED("thread.run.queued", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run moves to an in_progress status.
+     */
+    THREAD_RUN_IN_PROGRESS("thread.run.in_progress", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run moves to a requires_action status.
+     */
+    THREAD_RUN_REQUIRES_ACTION("thread.run.requires_action", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run is completed.
+     */
+    THREAD_RUN_COMPLETED("thread.run.completed", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run ends with status incomplete.
+     */
+    THREAD_RUN_INCOMPLETE("thread.run.incomplete", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run fails.
+     */
+    THREAD_RUN_FAILED("thread.run.failed", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run moves to a cancelling status.
+     */
+    THREAD_RUN_CANCELLING("thread.run.cancelling", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run is cancelled.
+     */
+    THREAD_RUN_CANCELLED("thread.run.cancelled", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run expires.
+     */
+    THREAD_RUN_EXPIRED("thread.run.expired", Run::class, Run.serializer()),
+
+    /**
+     * Occurs when a run step is created.
+     */
+    THREAD_RUN_STEP_CREATED("thread.run.step.created", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when a run step moves to an in_progress state.
+     */
+    THREAD_RUN_STEP_IN_PROGRESS("thread.run.step.in_progress", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when parts of a run step are being streamed.
+     */
+    THREAD_RUN_STEP_DELTA("thread.run.step.delta", RunStepDelta::class, RunStepDelta.serializer()),
+
+    /**
+     * Occurs when a run step is completed.
+     */
+    THREAD_RUN_STEP_COMPLETED("thread.run.step.completed", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when a run step fails.
+     */
+    THREAD_RUN_STEP_FAILED("thread.run.step.failed", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when a run step is cancelled.
+     */
+    THREAD_RUN_STEP_CANCELLED("thread.run.step.cancelled", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when a run step expires.
+     */
+    THREAD_RUN_STEP_EXPIRED("thread.run.step.expired", RunStep::class, RunStep.serializer()),
+
+    /**
+     * Occurs when a message is created.
+     */
+    THREAD_MESSAGE_CREATED("thread.message.created", Message::class, RunStep.serializer()),
```

Comment:
I think the serialised used here is the wrong one. I think it should be the Message one.
```suggestion
    THREAD_MESSAGE_CREATED("thread.message.created", Message::class, Message.serializer()),
```

Reply:
Good catch, fixed.